bolt-sidecar/src/state/execution.rs

```diff
@@ -543,11 +543,27 @@ impl<C: StateFetcher> ExecutionState<C> {
         self.block_templates.get(&slot)
     }
 
-    /// Gets the block template for the given slot number and removes it from the cache.
-    /// This should be called when we need to propose a block for the given slot,
-    /// or when a new head comes in which makes an older block template useless.
-    pub fn remove_block_template(&mut self, slot: u64) -> Option<BlockTemplate> {
-        self.block_templates.remove(&slot)
+    /// Removes all the block templates which slot is less then or equal `slot`, and returns them.
+    ///
+    /// This should be called when we need to propose a block for the given slot, or when a new
+    /// head comes in which makes an older block templates useless.
+    ///
+    /// NOTE: We remove all previous block templates to ensure that, when a new head is received
+    /// from the beacon client, all stale template are cleared. This prevents outdated templates
+    /// from persisting in cases of missed slots, where such events are not emitted.
+    pub fn remove_block_templates(&mut self, slot: u64) -> Vec<BlockTemplate> {
```

Comment:
what do you think of calling this `remove_block_templates_until` ?

Reply:
Gud one